Assignment 3 Instructions.Rmd

```diff
@@ -153,16 +153,161 @@ K7 <- dplyr::count(K4, cluster)
 
 Look at the number of people in each cluster, now repeat this process for 3 rather than 2 clusters. Which cluster grouping do you think is more informative? Write your answer below:
 
+```{r}
+# Put K3 into 3 clusters
+fit3 <- kmeans(K3,3)
+
+# to check whether the clustering has been done
+fit3$cluster
+
+# integrate the cluster information into the dataframe
+K31 <- data.frame(K3,fit3$cluster)
+
+# rename the vectors
+names(K31) <- c("1", "2", "3", "4", "5", "cluster")
+
+# transform the dataform from a wide from into a long one
+K32 <- gather(K31,key="week",value="motivation",-"cluster")
+
+# Group by week and cluster and then summarize the data
+K33 <- K32 %>% group_by(week, cluster) 
+K33 <- summarise(K33, avg = mean(motivation))
+
+# Change the class
+K33$week <- as.numeric(K33$week)
+K33$cluster <- as.factor(K33$cluster)
+
+# get the plot
+ggplot(K33, aes(week, avg, colour = cluster)) + geom_line() + xlab("Week") + ylab("Average Motivation")
+
+# Get the number of each plot
+K34 <- count(K31, cluster)
+```
 
 Once you have done this, save both of your plots to the Assignment 5 file. Create a Zotero item to index your new computer program (Assignment 5.rmd) in Zotero. Then commit your assignment, push it to your Github account and then Pull request your version to the original assignment version so I can see it.
 
 ##Part II
 
 Now, try to do the same for the data [collected in class](https://tccolumbia.qualtrics.com/SE/?SID=SV_6RRqlSAM6lZWYQt). Create two groups of clusters, the answers to the questions and regions where people grew up. 
 
+```{r}
+# read the data
+C1 <- read.csv("~/Desktop/github/assignment-3-YiranDuMirabelle11/cluster-class-data.csv")
+C2 <- select(C1, -c(1:3))
+
+#recode   Question2-- G-iff (1) & J-iff(0)
+C2[2] <- ifelse(C2[2]=="g-iff",1,0)
+
+#recode Question1-- Yes(1) NO(0)
+C2[1] <- ifelse(C2[1]=="Yes",1,0)
+
+# Get ride of the missing data
+C3 <- na.omit(C2)
+
+# Dataset of numeric answers
+C4 <- select(C3,-c(11:13))
+C5 <- select(C3,c(11:13))
+
+# GROUP 1- 2 clusters
+C4 <- scale(C4)
+classfit_2clu <- kmeans(C4,2)
+C41 <- data.frame(C4,classfit_2clu$cluster)
```

Comment:
You could also append the cluster to directly to your data frame without creating a new data frame and having to rename it.